app/controllers/projects_contestant_controller.rb

```diff
@@ -0,0 +1,14 @@
+class ProjectsContestantController < ApplicationController
+  def id
+    params[:id]
+  end
+  def new
```

Comment:
This wouldn't be a `new` action. This is the controller that builds the relationship between an existing project and an existing contestant. So it would look something like:
```ruby
def create
  @project = Project.find(params[:project_id])
  @contestant = Contestant.find(params[:contestant_id])
  ContestantProject.create!(project: @project, contestant: @contestant)

  redirect_to "/projects/#{@project.id}"
end
```
now this is going to also rely on your route. So the route on your form would need to point to this controller and action.
```ruby
post "/project/:project_id/contestants", to: "project_contestants#create"
```